src/remote_prune

```
@@ -0,0 +1,236 @@
+                except Exception as e:
+                    errors.append(e)
+            if snapshot_id and region_name:
+                try:
+                    delete_snapshot(snapshot_id, region=region_name)
+                except Exception as e:
+                    errors.append(e)
+
+    aliyun = build.images.get("aliyun")
+    if aliyun:
+        region_name = aliyun.get("name")
+        aliyun_id = aliyun.get("hvm")
+        if dry_run:
+            print(f"Will delete {aliyun_id} in {region_name}")
+        else:
+            if region_name and aliyun_id:
+                try:
+                    remove_aliyun_image(aliyun_id, region=region_name)
+                except Exception as e:
+                    errors.append(e)
+
+    azure = build.images.get("azure")
+    if azure:
+        image = azure.get("image")
+        resource_group = cloud_config.get("azure", {}).get("resource-group")
+        credentials = cloud_config.get("azure", {}).get("credentials")
+        if dry_run:
+            print(f"Will delete Azure {image}")
+        else:
+            if image and resource_group and credentials:
+                try:
+                    remove_azure_image(image, resource_group, credentials)
+                except Exception as e:
+                    errors.append(e)
+
+    gcp = build.images.get("gcp")
+    if gcp:
+        gcp_image = gcp.get("image")
+        json_key = cloud_config.get("gcp", {}).get("json-key")
+        project = cloud_config.get("gcp", {}).get("project")
+        if dry_run:
+            print(f"Will delete {gcp_image}")
+        else:
+            if gcp_image and json_key and project:
+                try:
+                    remove_gcp_image(gcp_image, json_key, project)
+                except Exception as e:
+                    errors.append(e)
+
+    if len(errors) != 0:
+        print(f"Found errors when removing build {build.id}:")
+        for e in errors:
+            print(e)
```

should we actually raise an exception here instead of just being informational?